investors/views.py

```diff
@@ -81,3 +84,42 @@ def destroy(self, request, pk=None):
             user_instance.save()
 
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @action(detail=True, methods=['get'], url_path='all_subscribed_projects')
+    def all_subscribed_projects(self, request, pk=None):
+        """
+        Get all subscribed projects of the investor.
+        This action retrieves a list of all projects that the investor is subscribed to.
+        Parameters:
+        - request (Request): The HTTP request object.
+        - pk (int): The primary key of the investor.
+        Returns:
+        Response: A JSON response containing a list of subscribed projects.
+        """
+        investor = get_object_or_404(Investor, id=pk)
+        subscribed_projects = investor.subscribed_projects.all()
+        serializer = ProjectSerializer(subscribed_projects, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
Please rename the URL path from the current one to `/api/investors/follows/` for better clarity and consistency.

Implement Logic for Displaying Investor's Projects: Update the logic to display all projects associated with the investor. If `pk=None`, the method should display all projects the current investor is subscribed to. (Check auth and role)

If pk is not provided (i.e., pk is None), the method should return all projects the user is subscribed to.
Otherwise, going to `/api/investors/1/follows/` the method should retrieve the investor using the provided pk and fetch their subscribed projects.
If `pk = None` and User is not Investor method should return an error

Reply:
Could you check my changes?